squash-honeycomb.ipynb

```diff
@@ -0,0 +1,381 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The SQuaSH `validate_drp.AM1` metric on a given dataset and filter can be obtained obtained from this query:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = {\n",
+    "    \"breakdowns\": [\n",
+    "        \"dataset\", \"filter_name\"\n",
+    "    ],\n",
+    "    \"calculations\": [\n",
+    "        {\"column\": \"validate_drp.AM1\", \"op\": \"AVG\"}\n",
+    "    ],\n",
+    "    \"filters\":[\n",
+    "      {\"column\": \"dataset\", \"op\": \"=\", \"value\": \"validation_data_hsc\"},\n",
+    "      {\"column\": \"filter_name\", \"op\": \"=\", \"value\": \"HSC-R\"}\n",
+    "    ],\n",
+    "    \"filter_combination\": \"AND\"\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Markers\n",
+    "Markers are annotations over the time series plot. Markers are defined per dataset and can be created programatically via the [Markers API](https://docs.honeycomb.io/api/markers/).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "headers = {'X-Honeycomb-Team': HONEY_API_KEY}\n",
+    "\n",
+    "marker={\"message\": \"Testing marker API\"}\n",
+    "\n",
+    "r = requests.post(\"https://api.honeycomb.io/1/markers/squash-demo\", json=marker, headers=headers)\n",
+    "r.json()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All the Markers in a dataset may be retrieved by:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = requests.get(\"https://api.honeycomb.io/1/markers/squash-demo\", headers=headers)\n",
+    "r.json()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Triggers\n",
+    "\n",
+    "Given a time series and a threshold one can use the [Triggers API](https://docs.honeycomb.io/api/triggers/) for setting alerts when the values pass a threshold. Triggers are also defined per dataset. Honeycomb provides integration with Slack for alert notification. \n",
+    "\n",
+    "Let's set an alert for the `validate_drp.AM1` metric. We'll get the design specification for that metric from the SQuaSH API:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spec = requests.get(SQUASH_API_URL + \"spec/validate_drp.AM1.design\" ).json()\n",
+    "spec"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trigger = {\"name\": \"AM1 alert\",\n",
+    "           \"query\": query,\n",
+    "           \"threshold\": {\"op\": spec[\"threshold\"][\"operator\"], \n",
```

Comment:
It's really cool that the threshold conforms to our notion of a threshold so well.  I worry, though, that some of our operators are `<=`.  Do you know if those operators are valid in trigger API?  The docs were not explicit about what operators are allowed.

Reply:
I agree it is not evident from the docs. In the Triggers API documentation they say "Filter op" which is documented elsewhere https://docs.honeycomb.io/api/query-specification/#filter-operators

Also, the operator in the `lsst.verify` specifications is such that  measurement `op` spec is True if the measurement passes the spec. But we need the inverse operator when configuring the alerts.

I'll include a note about that and an inverse mapping in the notebook.